mkosi/qemu.py

```diff
@@ -679,7 +728,26 @@ def finalize_firmware_variables(
             if config.qemu_firmware_variables == Path("microsoft") or not config.qemu_firmware_variables
             else config.qemu_firmware_variables
         )
-        shutil.copy2(vars, Path(ovmf_vars.name))
+        if config.secure_boot_certificate:
+            run(
+                [
+                    "virt-fw-vars",
+                    "--input", vars,
+                    "--output", ovmf_vars.name,
+                    "--set-json", ovmf_json.name,
+                    "--loglevel", "WARNING",
+                ],
+                sandbox=config.sandbox(
+                    binary=qemu,
+                    options=[
+                        "--bind", ovmf_vars.name, ovmf_vars.name,
+                        "--ro-bind", ovmf_json.name, ovmf_json.name,
+                        "--ro-bind", vars, vars,
+                    ],
+                ),
+            )
+        else:
+            shutil.copy2(vars, Path(ovmf_vars.name))
```

Comment:
This should be dropped, just because the image is built with a secure boot certificate shouldn't imply that we automatically add it to any given OVMF variables, especially ones provided by the user.

Reply:
it's now skipped if there are user-provided vars